File: server/src/main/java/com/waridley/chatgame/server/CommandExecutive.java
package com.waridley.chatgame.server;

import com.waridley.chatgame.api.backend.GameStorageInterface;
import com.waridley.chatgame.game.Player;
import com.waridley.ttv.TtvStorageInterface;
import com.waridley.ttv.TtvUser;
import lombok.AllArgsConstructor;

public class CommandExecutive {
	
	private final TtvStorageInterface ttvStorageInterface;
	private final GameStorageInterface gameStorageInterface;
	
	final InfoRelayer info;
	final PermissionChecker check;
	
	CommandExecutive(TtvStorageInterface ttvStorageInterface, GameStorageInterface gameStorageInterface) {
		this.ttvStorageInterface = ttvStorageInterface;
		this.gameStorageInterface = gameStorageInterface;
		this.info = new InfoRelayer(ttvStorageInterface, gameStorageInterface);
		this.check = new PermissionChecker(ttvStorageInterface, gameStorageInterface);
	}
	
	
	@AllArgsConstructor
	static class InfoRelayer {
		
		private TtvStorageInterface ttvStorageInterface;
		private GameStorageInterface gameStorageInterface;
		
		TtvUser ttvUserFromLogin(String login) {
			return ttvStorageInterface.findOrCreateTtvUser(login);
		}
		
		Player playerFromTtvUser(TtvUser ttvUser) {
			return gameStorageInterface.findOrCreatePlayer(ttvUser);
		}
		
		Player playerFromTtvUserId(String userId) {
			return gameStorageInterface.findOrCreatePlayer(Long.parseLong(userId));
		}
	}
	
	@AllArgsConstructor
	static class PermissionChecker {
		
		private TtvStorageInterface ttvStorageInterface;
		private GameStorageInterface gameStorageInterface;
		
		
		
	}
	
	
	
}

File: server/src/main/java/com/waridley/chatgame/server/CommandExecutive.kt
package com.waridley.chatgame.server

import com.waridley.chatgame.api.backend.GameStorageInterface
import com.waridley.chatgame.game.Player
import com.waridley.ttv.TtvStorageInterface
import com.waridley.ttv.TtvUser

class CommandExecutive internal constructor(private val ttvStorageInterface: TtvStorageInterface, private val gameStorageInterface: GameStorageInterface) {
	@JvmField
	val info: InfoRelayer
	val check: PermissionChecker
	
	class InfoRelayer(private val ttvStorageInterface: TtvStorageInterface, private val gameStorageInterface: GameStorageInterface) {
		fun ttvUserFromLogin(login: String?): TtvUser {
			return ttvStorageInterface.findOrCreateTtvUserFromLogin(login)
		}
		
		fun playerFromTtvUser(ttvUser: TtvUser): Player? {
			return gameStorageInterface.findOrCreatePlayer(ttvUser)
		}
		
		fun playerFromTtvUserId(userId: String): Player? {
			return gameStorageInterface.findOrCreatePlayerByTtvId(userId)
		}
		
	}
	
	class PermissionChecker(private val ttvStorageInterface: TtvStorageInterface, private val gameStorageInterface: GameStorageInterface)
	
	init {
		info = InfoRelayer(ttvStorageInterface, gameStorageInterface)
		check = PermissionChecker(ttvStorageInterface, gameStorageInterface)
	}
}